File: Models/Brand.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace FirstCoreAngularTry.Models
{
    public partial class Brand
    {
        public Brand()
        {
            Cpus = new HashSet<Cpu>();
            GraphicCardModels = new HashSet<GraphicCardModel>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public bool Archived { get; set; }

        public virtual ICollection<Cpu> Cpus { get; set; }
        public virtual ICollection<GraphicCardModel> GraphicCardModels { get; set; }
    }
}

File: ClientApp/src/app/brand/brand.ts
import { Cpu } from "../cpu/cpu";
import { GraphicCardModel } from "../graphic-card-model/graphic-card-model";

export interface Brand {
  id: number;
  name: string;
  archived: boolean;
  Cpus: Cpu[];
  GraphicCardModels: GraphicCardModel[]
}
